test/math.jl

```diff
@@ -652,14 +652,15 @@ end
 end
 
 @testset "modf" begin
-    @testset "$elty" for elty in (Float16, Float32, Float64)
-        @test modf( convert(elty,1.2) )[1] ≈ convert(elty,0.2)
-        @test modf( convert(elty,1.2) )[2] ≈ convert(elty,1.0)
-        @test modf( convert(elty,1.0) )[1] ≈ convert(elty,0.0)
-        @test modf( convert(elty,1.0) )[2] ≈ convert(elty,1.0)
-        @test isequal(modf( convert(elty,-Inf) ), (-0.0, -Inf))
-        @test isequal(modf( convert(elty,Inf) ), (0.0, Inf))
-        @test isequal(modf( convert(elty,NaN) ), (NaN, NaN))
+    @testset "$T" for T in (Float16, Float32, Float64)
+        @test modf(T(1.2))[1] ≈ T(0.2)
```

Comment:
Shouldn't these be exactly equal? I'm not sure why they were approximate before.

Reply:
I was wondering the same. I can change that as part of this PR.

Reply:
On master
```
  Expression: modf(T(1.2)) == (T(0.2), T(1.0))
   Evaluated: (0.19999999999999996, 1.0) == (0.2, 1.0)
```

Reply:
then maybe add a test for `@test modf(T(1.25))[1] == T(0.25)` since that one should always hold.